### ISSUE TITLE
travis lint: unexpected keys jobs, stages

### ISSUE THREAD

**Issue by axnsan12** [travis-ci/travis.rb#571]
When linting my `.travis.yml` with top-level `jobs` and `stages`, I get two invalid warnings from `travis lint`:

```
Warnings for .travis.yml:
[x] unexpected key jobs, dropping
[x] unexpected key stages, dropping
```

The configuration works just fine in CI: https://travis-ci.org/axnsan12/drf-yasg/builds/322223668

**Comment by sigo** [comment]
Travis stage builds is beta feature. Maybe this is the reason? I have the same situation.

**Comment by sigo** [comment]
[Build stages are beta feature.](https://github.com/cloudflare/unsee/pull/213#issuecomment-359228732) We need just wait.

**Comment by joar** [comment]
> @sigo [Build stages are beta feature.](https://github.com/cloudflare/unsee/pull/213#issuecomment-359228732) We need just wait.

[Build stages are GA since July](https://blog.travis-ci.com/2018-07-18-build-stages-officially-released)

**Comment by pitpit** [comment]
Still having this issue :

```
$ travis --version
1.8.9
$ travis lint
Warnings for .travis.yml:
[x] unexpected key stages, dropping
[x] unexpected key jobs, dropping
```

**Comment by tkornut** [comment]
I am trying to learn how to use travis command line tool, not using stages. 
Configuration file:
https://github.com/IBM/mi-toolchain/blob/master/.travis.yml
is working properly on Travis:
https://travis-ci.com/IBM/mi-toolchain
but with travis cli I got the same:
`
$ travis --version
1.8.9
$ travis lint .travis.yml
Warnings` for .travis.yml:
[x] unexpected key jobs, dropping
`

**Comment by DrLuke** [comment]
This is still throwing incorrect warnings:
````
$ travis lint .travis.yml 
Warnings for .travis.yml:
[x] unexpected key stages, dropping
[x] unexpected key jobs, dropping
$ travis version
1.8.9

````

**Comment by paulkatsoulakis** [comment]
> This is still throwing incorrect warnings:
> 
> ```
> $ travis lint .travis.yml 
> Warnings for .travis.yml:
> [x] unexpected key stages, dropping
> [x] unexpected key jobs, dropping
> $ travis version
> 1.8.9
> ```

hello, same here

**Comment by ConnorBP** [comment]
bump... any update on this? The linter really isn't being reliable. From what I gather my file might actually be fine and the linter is spitting out wrong errors? I really can't tell until I push this .travis.yml to github

```shell
macbook-pro:myproject Connor$ travis lint --no-interactive .travis.yml
Warnings for .travis.yml:
[x] unexpected key jobs, dropping
[x] unexpected key stages, dropping
[x] value for cache section is empty, dropping
[x] in cache section: unexpected key cargo, dropping
```

**Comment by pixelicous** [comment]
Same here, my editor also doesn't like me adding stages and matrix include
Below parts of .travis.yml 
```yml
jobs:
  stages:
  - name: x
  include:
    matrix:
    - stage: x
```

>  $ travis lint
> Warnings for .travis.yml:
> [x] unexpected key jobs, dropping

> 
> travis --version
> 1.8.10

**Comment by rwenz3l** [comment]
TL;DR: Use [config.travis-ci.org](https://config.travis-ci.org/)


Alright, I wanted to rant.. but that does not help the issue. For anyone stumbling here in late 2019:
The linter is still broken and will remain so until the underlying yaml library (travis-yaml) has been replaced by the new one (travis-yml) .. it's a big rewrite and takes a lot of time, so yeah.. forget the linter for now.

Status as of 08/2019:
- [lint.travis-ci.org](https://lint.travis-ci.org/) is deprecated/broken
- `travis lint` (from this repo) is partially broken (does not detect certain keys)
- [api.travis-ci.com/lint](https://api.travis-ci.com/lint) is partially broken (does not detect certain keys)

After some digging in all the Issues and Repositories, I found the travis-ci config parser
https://config.travis-ci.org/, which seems to work so far.

**Comment by BanzaiMan** [comment]
The `lint` command has gone through substantial changes recently. If you still see this problem, please open a new issue.
